To: vim_dev@googlegroups.com
Subject: Patch 7.3.057
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.057
Problem:    Segfault with command line abbreviation. (Randy Morris)
Solution:   Don't retrigger the abbreviation when abandoning the command line.
            Continue editing the command line after the error.
Files:      src/ex_getln.c
    

*** ../vim-7.3.056/src/ex_getln.c	2010-11-10 15:37:00.000000000 +0100
--- src/ex_getln.c	2010-11-16 14:03:09.000000000 +0100
***************
*** 712,718 ****
  		    }
  		}
  		beep_flush();
! 		c = ESC;
  	    }
  #endif
  	    else
--- 712,722 ----
  		    }
  		}
  		beep_flush();
! 		got_int = FALSE;	/* don't abandon the command line */
! 		did_emsg = FALSE;
! 		emsg_on_display = FALSE;
! 		redrawcmd();
! 		goto cmdline_not_changed;
  	    }
  #endif
  	    else
*** ../vim-7.3.056/src/version.c	2010-11-16 11:29:30.000000000 +0100
--- src/version.c	2010-11-16 14:04:25.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     57,
  /**/

-- 
FROG: How you English say:  I one more time, mac, I unclog my nose towards
      you, sons of a window-dresser,  so, you think you could out-clever us
      French fellows with your silly knees-bent creeping about advancing
      behaviour.  (blows a raspberry) I wave my private parts at your aunties,
      you brightly-coloured, mealy-templed, cranberry-smelling, electric
      donkey-bottom biters.
                 "Monty Python and the Holy Grail" PYTHON (MONTY) PICTURES LTD

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
